Scenes/Player/MapGenerator.gd

```diff
@@ -0,0 +1,12 @@
+extends TileMapLayer
+# --- Exported Properties ---
+func _ready() -> void:
+	global.pathfinder = AStarGrid2D.new()
```

Comment:
Note: take a look at the NavigationAgent2D node and tilemap-based navigation solutions, they're much more streamlined. Only issue is they're a bit finicky when it comes to dynamic updating.

Reply:
yes, but for the interaction system, preformence, and mechanical, reasons it is better to use ASTAR PATHFINDINGGrid2D as it requires much less cpu and ram with the features we want built in. unless you have a solution for every one of these problems, I feel it would be more effective to continue using the built in grid pathfinding.

Reply:
okay curiosity git the best of me, why do you think nav agent 2d's are more performent?

Reply:
It's less about performance and more future proofing and simplicity, as the nav agent system is just an abstraction of the AStar grid system. No worries for now.